File: pset6/readability/readability.py
from cs50 import get_string

sentence = get_string("Text: ")  # asking for input and some variables
period = 0
letters = 0
words = 1
for i in range(len(sentence)):  # counting the ends of sentences
    if sentence[i] == "." or sentence[i] == "!" or sentence[i] == "?":
        period += 1
for i in range(len(sentence)):  # counting the words in my sentence
    if sentence[i] == " ":
        words += 1
for i in range(len(sentence)):  # counting my letters in my sentence
    if sentence[i].isalnum():
        letters += 1
L = float((letters * 100) / words)  # made both floats to prevent mistakes
S = float((period * 100) / words)
index = round(0.0588 * L - 0.296 * S - 15.8)  # formula that is given
if index < 1:  # depending on the number the grade will be printed
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade", index)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string sentence = get_string("Text: "); // gets the sentence and initializes some variables
    int periods = 0;
    int letters = 0;
    int words = 0;
    for (int i = 0; i < strlen(sentence); i++) // finds the number of sentences
    {
        if (sentence[i] == '.' || sentence[i] == '!' || sentence[i] == '?')
        {
            periods ++;
        }
    }
    for (int i = 0; i < strlen(sentence); i++) // finds the number of letters
    {
        if (isalpha(sentence[i]))
        {
            letters ++;
        }
    }
    words = 1;
    for (int i = 0; i < strlen(sentence); i++) // finds the number of words
    {
        if (isspace(sentence[i]))
        {
            words ++;
        }
    }
    float L = (letters * 100) / (float)words;
    float S = (periods * 100) / (float)words;
    float index = 0.0588 * L - 0.296 * S - 15.8; // plugs the variables into the formula
    int index2 = round(index);
    if (index2 < 1) // figures out what to print depending on the Grade level
    {
        printf("Before Grade 1\n");
    }
    else if (index2 > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index2);
    }
}